assets/cluster-monitoring-operator/prometheus-rule.yaml

```
@@ -28,6 +28,37 @@ spec:
       for: 15m
       labels:
         severity: warning
+    - alert: HighlyAvailableWorkloadIncorrectlySpread
+      annotations:
+        description: Workload {{ $labels.namespace }}/{{ $labels.workload }} is incorrectly
+          spread across multiple nodes which breaks high-availability requirements.
+          There are {{ $value }} pods on node {{ $labels.node }}, where there should
+          only be one. Since the workload is using persistent volumes, manual intervention
+          is needed. Please follow the guidelines provided in the runbook of this
+          alert to fix this issue.
+        runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/HighlyAvailableWorkloadIncorrectlySpread.md
+        summary: Highly-available workload is incorrectly spread across multiple nodes
+          and munual intervention is needed.
+      expr: |
+        count by(node,workload,namespace)
+          (
+            kube_pod_info{node!=""}
+            * on(namespace,pod) group_left()
```

```suggestion
            * on(namespace,pod) group_left(workload)
```

Without the above `workload` label is removed and `count by(node,workload,namespace)` cannot work correctly.